surefire-api/src/test/java/org/apache/maven/surefire/api/stream/AbstractStreamDecoderTest.java

```diff
@@ -258,6 +258,56 @@ public void shouldReadString() throws Exception
             .isEqualTo( "0123456789" );
     }
 
+    @Test
+    public void shouldReadStringOverflowOnNewLine() throws Exception
+    {
+        StringBuilder s = new StringBuilder( 1025 );
+        for ( int i = 0; i < 10; i++ )
+        {
+            s.append( PATTERN1 );
+        }
+        s.append( PATTERN1, 0, 23 );
+        s.append( "\u00FA\n" ); // 2-bytes encoded character + LF
+
+        Channel channel = new Channel( s.toString().getBytes( UTF_8 ), s.length() );
+
+        Mock thread = new Mock( channel, new MockForkNodeArguments(),
+            Collections.<Segment, ForkedProcessEventType>emptyMap() );
+
+        Memento memento = thread.new Memento();
+
+        assertThat( (String) invokeMethod( thread, "readString", memento, 1026 ) )
+            .isEqualTo( s.toString() );
+
+        assertThat ( memento.getByteBuffer().remaining() )
+            .isEqualTo( 0 );
+    }
+
+    @Test
+    public void shouldReadStringOverflowOn4BytesEncodedSymbol() throws Exception
+    {
+        StringBuilder s = new StringBuilder( 1024 );
```

Comment:
Just realized this should be 1025 here, 😵 (`"\uD83D\uDE35"`) is actually counted as 2 characters.

Do you want me to change this or just leave StringBuilder to increase the capacity on it's own?

Reply:
I don't know. The point of the unit tests is that you provide them and I would be able to reproduce a test failure with the original code. And I would apply your fix which makes the test green again. This was the process on my side.
It;s up to you. The tests are just a kind of language you as contributor want to convince ASF about the whole point of the fix. If it is necessary to write more tests, feel free to do it but then I will undergo the test with verifying failure on the original master code.

I found your interesting IT
https://github.com/zoltanmeze/surefire-corrupted-channel

Reply:
I will verify the code with/out the fix but I think the same as you. The unit tests are quite clear.
If there is only a problem in `AbstractStreamDecoder` and nowhere else, the IT is useless.